[Bug]: Unremoved deprecated code in MemoryProtectionExceptionHandlerLib

### Is there an existing issue for this?

- [x] I have searched existing issues

### Current Behavior

MemoryProtectionExceptionHandlerLib can not be compiled(aarch64) due to the deprecated and removed file "MemoryProtectionNonstopMode.h".
https://github.com/microsoft/mu_plus/blob/f7c042a9d1428cdf6be773e2b8ca6cf51f66cb6f/MsCorePkg/Library/MemoryProtectionExceptionHandlerLib/MemoryProtectionExceptionHandlerLibAArch64.c#L18

Log:
```
INFO - Building ... MU_BASECORE/MdeModulePkg/Library/BaseExceptionPersistenceLibNull/BaseExceptionPersistenceLibNull.inf [AARCH64]
INFO - Common/MU/MsCorePkg/Library/MemoryProtectionExceptionHandlerLib/MemoryProtectionExceptionHandlerLibAArch64.c:18:10: fatal error: 'Protocol/MemoryProtectionNonstopMode.h' file not found
INFO -    18 | #include <Protocol/MemoryProtectionNonstopMode.h>
INFO -       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Expected Behavior

This include line should be removed and a functionality check is required.

### Steps To Reproduce

Compile with this lib when building AARCH64 target => lack the header => BOOM.

### Build Environment

```markdown
- OS(s): Ubuntu 24.04
- Tool Chain(s): CLANGPDB
- Targets Impacted: ALL
- Target Arch: AARCH64
```

### Version Information

```text
commit: f7c042a9d1428cdf6be773e2b8ca6cf51f66cb6f
```

### Urgency

Medium

### Are you going to fix this?

Someone else needs to fix it

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

_No response_

--- a/.azurepipelines/Ubuntu-CLANGPDB.yml
+++ b/.azurepipelines/Ubuntu-CLANGPDB.yml
@@ -1,0 +1,69 @@
+## @file
+# Azure Pipeline build file for a build using Ubuntu and CLANGPDB.
+#
+# IMPORTANT: This file requires the local repo to contain a .azurepipelines/Matrix-Build-Job.yml file that
+#            performs the actual build steps for the configurations and packages supported by the repo.
+#
+# NOTE: This file is automatically synchronized from Mu DevOps. Update the original file there
+#       instead of the file in this repo.
+#
+# - Mu DevOps Repo: https://github.com/microsoft/mu_devops
+# - File Sync Settings: https://github.com/microsoft/mu_devops/blob/main/.sync/Files.yml
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+variables:
+- group: architectures-arm64-x86-64
+- group: tool-chain-ubuntu-clangpdb
+- group: coverage
+
+extends:
+  template: MuDevOpsWrapper.yml
+  parameters:
+    do_ci_build: true
+    do_ci_setup: true
+    do_non_ci_build: false
+    do_non_ci_setup: true
+    do_pr_eval: true
+    container_build: true
+    rust_build: true
+    os_type: Linux
+    build_matrix:
+      TARGET_MFCI_XML:
+        Pkgs: 'XmlSupportPkg,MfciPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_GRAPHICS_WHEA_HID:
+        Pkgs: 'MsGraphicsPkg,MsWheaPkg,HidPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_CORE:
+        Pkgs: 'MsCorePkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_TEST_PCBDS_LOG:
+        Pkgs: 'TpmTestingPkg,UefiTestingPkg,PcBdsPkg,AdvLoggerPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_MFCI_XML_ARM:
+        Pkgs: 'XmlSupportPkg,MfciPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true
+      TARGET_GRAPHICS_WHEA_HID_ARM:
+        Pkgs: 'MsGraphicsPkg,MsWheaPkg,HidPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true
+      TARGET_CORE_ARM:
+        Pkgs: 'MsCorePkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true
+      TARGET_TEST_PCBDS_LOG_ARM:
+        Pkgs: 'TpmTestingPkg,UefiTestingPkg,PcBdsPkg,AdvLoggerPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true

--- a/.azurepipelines/Windows-CLANGPDB.yml
+++ b/.azurepipelines/Windows-CLANGPDB.yml
@@ -1,0 +1,68 @@
+## @file
+# Azure Pipeline build file for a build using Windows and the CLANG PDB toolchain.
+#
+# IMPORTANT: This file requires the local repo to contain a .azurepipelines/Matrix-Build-Job.yml file that
+#            performs the actual build steps for the configurations and packages supported by the repo.
+#
+# NOTE: This file is automatically synchronized from Mu DevOps. Update the original file there
+#       instead of the file in this repo.
+#
+# - Mu DevOps Repo: https://github.com/microsoft/mu_devops
+# - File Sync Settings: https://github.com/microsoft/mu_devops/blob/main/.sync/Files.yml
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+variables:
+- group: architectures-x86-64
+- group: tool-chain-windows-clangpdb
+- group: coverage
+
+extends:
+  template: MuDevOpsWrapper.yml
+  parameters:
+    do_ci_build: true
+    do_ci_setup: true
+    do_non_ci_build: false
+    do_non_ci_setup: true
+    do_pr_eval: true
+    rust_build: true
+    os_type: Windows_NT
+    build_matrix:
+      TARGET_MFCI_XML:
+        Pkgs: 'XmlSupportPkg,MfciPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_GRAPHICS_WHEA_HID:
+        Pkgs: 'MsGraphicsPkg,MsWheaPkg,HidPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_CORE:
+        Pkgs: 'MsCorePkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_TEST_PCBDS_LOG:
+        Pkgs: 'TpmTestingPkg,UefiTestingPkg,PcBdsPkg,AdvLoggerPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+      TARGET_MFCI_XML_ARM:
+        Pkgs: 'XmlSupportPkg,MfciPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true
+      TARGET_GRAPHICS_WHEA_HID_ARM:
+        Pkgs: 'MsGraphicsPkg,MsWheaPkg,HidPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true
+      TARGET_CORE_ARM:
+        Pkgs: 'MsCorePkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true
+      TARGET_TEST_PCBDS_LOG_ARM:
+        Pkgs: 'TpmTestingPkg,UefiTestingPkg,PcBdsPkg,AdvLoggerPkg'
+        Targets: 'DEBUG,RELEASE'
+        ArchList: $(arch_list)
+        SelfHostAgent: true

--- a/AdvLoggerPkg/AdvLoggerOsConnectorPrm/AdvLoggerOsConnectorPrm.inf
+++ b/AdvLoggerPkg/AdvLoggerOsConnectorPrm/AdvLoggerOsConnectorPrm.inf
@@ -38,5 +38,8 @@
   MSFT:*_*_*_DLINK_FLAGS  = /DLL /SUBSYSTEM:CONSOLE /VERSION:1.0
   MSFT:*_*_*_GENFW_FLAGS = --keepoptionalheader
 
+  CLANGPDB:*_*_*_DLINK_FLAGS  = /DLL /SUBSYSTEM:CONSOLE /VERSION:1.0
+  CLANGPDB:*_*_*_GENFW_FLAGS = --keepoptionalheader
+
   GCC:*_*_AARCH64_GENFW_FLAGS = --keepoptionalheader --prm
   GCC:*_*_AARCH64_DLINK_FLAGS = -Wl,--no-gc-sections -Wl,--require-defined=PrmModuleExportDescriptor -Wl,--require-defined=AdvLoggerOsConnectorPrmHandler

--- a/MsGraphicsPkg/Application/BmpDisplay/BmpDisplay.c
+++ b/MsGraphicsPkg/Application/BmpDisplay/BmpDisplay.c
@@ -169,6 +169,7 @@ BmpDisplayEntrypoint (
   CursorModified          = FALSE;
   BmpFileData             = NULL;
   OriginalVideoBufferData = NULL;
+  Blt                     = NULL;
 
   Status = ParseCommandLine (&BmpFilePath);
   if (EFI_ERROR (Status) || (BmpFilePath == NULL)) {
@@ -253,7 +254,6 @@ BmpDisplayEntrypoint (
   //
   // Translate the GOP image buffer to a BLT buffer
   //
-  Blt         = NULL;
   ImageWidth  = 0;
   ImageHeight = 0;
   Status      =  TranslateBmpToGopBlt (

--- a/XmlSupportPkg/Test/UnitTest/XmlTreeLib/XmlTreeLibUnitTests.c
+++ b/XmlSupportPkg/Test/UnitTest/XmlTreeLib/XmlTreeLibUnitTests.c
@@ -279,9 +279,9 @@ ParseInValidXml1 (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData = NULL;
-  EFI_STATUS  Status;
-  CHAR8       BadString[] = "This is not valid xml"; // Not even close
+  XmlNode      *ResultData = NULL;
+  EFI_STATUS   Status;
+  CONST CHAR8  *BadString = "This is not valid xml";  // Not even close
 
   Status = CreateXmlTree (BadString, AsciiStrLen (BadString), &ResultData);
   if (ResultData != NULL) {
@@ -303,9 +303,9 @@ ParseInValidXml2 (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData = NULL;
-  EFI_STATUS  Status;
-  CHAR8       BadString[] = "<Node1><Node2></Node1>";  // malformed XML
+  XmlNode      *ResultData = NULL;
+  EFI_STATUS   Status;
+  CONST CHAR8  *BadString = "<Node1><Node2></Node1>";   // malformed XML
 
   Status = CreateXmlTree (BadString, AsciiStrLen (BadString), &ResultData);
   if (ResultData != NULL) {
@@ -327,9 +327,9 @@ ParseInValidXml3 (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData = NULL;
-  EFI_STATUS  Status;
-  CHAR8       BadString[] = "<Node1><Node2><Node3 /></Node1>";  // malformed xml - Missing Node 2 closure
+  XmlNode      *ResultData = NULL;
+  EFI_STATUS   Status;
+  CONST CHAR8  *BadString = "<Node1><Node2><Node3 /></Node1>";   // malformed xml - Missing Node 2 closure
 
   Status = CreateXmlTree (BadString, AsciiStrLen (BadString), &ResultData);
   if (ResultData != NULL) {
@@ -351,10 +351,10 @@ TestNodeCount (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData   = NULL;
-  UINTN       TotalElements = 0;
-  EFI_STATUS  Status;
-  CHAR8       MyString[] = "<Node1><Node2><Node3 /><Node4 /></Node2> </Node1>"; // 4 nodes with depth of 3
+  XmlNode      *ResultData   = NULL;
+  UINTN        TotalElements = 0;
+  EFI_STATUS   Status;
+  CONST CHAR8  *MyString = "<Node1><Node2><Node3 /><Node4 /></Node2> </Node1>";  // 4 nodes with depth of 3
 
   Status = CreateXmlTree (MyString, AsciiStrLen (MyString), &ResultData);
   UT_ASSERT_NOT_NULL (ResultData);
@@ -378,10 +378,10 @@ TestNodeMaxDepth (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData = NULL;
-  UINTN       MaxDepth    = 0;
-  EFI_STATUS  Status;
-  CHAR8       MyString[] = "<Node1><Node2><Node3 /><Node4 /></Node2><Node5><Node6><Node7 /></Node6></Node5></Node1>"; // 4 nodes with depth of 4
+  XmlNode      *ResultData = NULL;
+  UINTN        MaxDepth    = 0;
+  EFI_STATUS   Status;
+  CONST CHAR8  *MyString = "<Node1><Node2><Node3 /><Node4 /></Node2><Node5><Node6><Node7 /></Node6></Node5></Node1>";  // 4 nodes with depth of 4
 
   Status = CreateXmlTree (MyString, AsciiStrLen (MyString), &ResultData);
   UT_ASSERT_NOT_NULL (ResultData);
@@ -405,10 +405,10 @@ TestAttributeCount (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData = NULL;
-  UINTN       Count       = 0;
-  EFI_STATUS  Status;
-  CHAR8       MyString[] = "<Node1 att1='test1'><Node2 att2='test2'><Node3 att3='test3' att4='test4'  /></Node2></Node1>"; // 4 attributes total
+  XmlNode      *ResultData = NULL;
+  UINTN        Count       = 0;
+  EFI_STATUS   Status;
+  CONST CHAR8  *MyString = "<Node1 att1='test1'><Node2 att2='test2'><Node3 att3='test3' att4='test4'  /></Node2></Node1>";  // 4 attributes total
 
   Status = CreateXmlTree (MyString, AsciiStrLen (MyString), &ResultData);
   UT_ASSERT_NOT_NULL (ResultData);
@@ -432,10 +432,10 @@ TestAttributeMax (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  XmlNode     *ResultData = NULL;
-  UINTN       Count       = 0;
-  EFI_STATUS  Status;
-  CHAR8       MyString[] = "<Node1 att1='test1'><Node2 att2='test2'><Node3 att3='test3' att4='test4'  /></Node2></Node1>"; // 4 attributes total
+  XmlNode      *ResultData = NULL;
+  UINTN        Count       = 0;
+  EFI_STATUS   Status;
+  CONST CHAR8  *MyString = "<Node1 att1='test1'><Node2 att2='test2'><Node3 att3='test3' att4='test4'  /></Node2></Node1>";  // 4 attributes total
 
   Status = CreateXmlTree (MyString, AsciiStrLen (MyString), &ResultData);
   UT_ASSERT_NOT_NULL (ResultData);

